jerry-core/ecma/builtin-objects/ecma-builtin-bigint.c

```diff
@@ -36,6 +52,358 @@
  * @{
  */
 
+/**
+ * The BigInt object's 'asIntN' and 'asUintN' routines
+ *
+ * See also:
+ *          ECMA-262 v5, 11.0
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_bigint_object_as_int_n (ecma_value_t this_arg, /**< 'this' argument */
+                                     ecma_value_t bits, /**< number of bits */
+                                     ecma_value_t bigint, /**< bigint number */
+                                     bool is_signed) /**< The operation is signed */
+{
+  JERRY_UNUSED (this_arg);
+
+  ecma_number_t input_bits;
+  ecma_value_t bit_value = ecma_op_to_index (bits, &input_bits);
+
+  if (ECMA_IS_VALUE_ERROR (bit_value))
+  {
+    return bit_value;
+  }
+
+  ecma_value_t bigint_value = ecma_bigint_to_bigint (bigint, false);
+
+  if (ECMA_IS_VALUE_ERROR (bigint_value))
+  {
+    return bigint_value;
+  }
+
+  if (input_bits == 0 || bigint_value == ECMA_BIGINT_ZERO)
+  {
+    ecma_free_value (bigint_value);
+    return ECMA_BIGINT_ZERO;
+  }
+
+  const uint32_t four_byte = (sizeof (uint32_t) * sizeof (uint64_t));
+  const uint32_t one_byte = sizeof (uint64_t);
+
+  uint32_t whole_part = (ecma_value_t) input_bits / four_byte;
```

Comment:
input_bits is whole number between [0- 2^53 - 1]. Division by  `four_byte` results a number between [0- 2^48 - 1] which cannot fit into an uint32_t, and also the ecma_value_t cast is invalid.

Reply:
uint64_t whole_part = floor (input_bits / 32);